File: C/2839.c
#include<stdio.h>

int main()
{
	int input;
	scanf("%d", &input);
	
	int out5 = input / 5;
	int count = 0;
	int out5_n = 0;
	int min_num = 0;
	for(;count <= out5;count ++)
	{
		out5_n = input - (count * 5);
		if((out5_n % 3) != 0)
		{
			continue;
		}
		else
		{
			min_num = count + (out5_n / 3);	
		}
	}
	if(min_num == 0)
	{
		min_num = -1;
	}
	printf("%d", min_num);
	return 0;
}

File: PythonBOJ/2839.py
a = input()

test = 0
min_num = 0  
a_n = 0

while(test <= (int(a) // 5)):
    a_n = int(a) - (test *5) 
    if((a_n % 3) != 0):
        test += 1
        continue
    else:
        min_num = test + (a_n / 3)
        test += 1

if(min_num == 0):
    min_num = -1

print(int(min_num))

    
        
        
